asset/asset.go

```diff
@@ -426,8 +427,17 @@ type GroupKey struct {
 	//   groupInternalKey + sha256(groupInternalKey || genesisOutPoint) * G
 	GroupPubKey btcec.PublicKey
 
-	// Sig is a signature over an asset's ID by `Key`.
-	Sig schnorr.Signature
+	// TapscriptRoot is the root of the Tapscript tree that commits to all
+	// script spend conditions for the group key. Instead of spending an
+	// asset, these scripts are used to define witnesses more complex than
+	// a Schnorr signature for reissuing assets.
+	TapscriptRoot [sha256.Size]byte
```

Comment:
Important to note that this can be empty (key-spend only output key). Should we make this nillable, or is it good enough to have it be all zero?

Reply:
Good point. Turned it into a `[]byte` with a comment. Can do some additional validation later on.